src/shared/components/RichTable.tsx

```diff
@@ -593,6 +627,73 @@ export const RichTable = <D extends object = {}>(config: RichTableProps<D>) => {
     config.bagRef(bag);
   }
 
+  const Pagination = () => {
+    const [displayPageSelector, setDisplayPageSelector] = useState(false);
+    const pageSelectorRef = useRef<FormControl & HTMLInputElement>(null);
+    const pageSelectorId = useRef(`RichTable_pageSelector_${Math.random()}`);
+
+    useLayoutEffect(() => {
+      if (displayPageSelector) {
+        pageSelectorRef.current?.focus();
+      }
+    }, [displayPageSelector]);
+
+    if (displayPageSelector) {
+      return (
+        <Col lg={2} className="RichTable_pagination ml-lg-auto justify-content-center justify-content-lg-end">
+          <Form.Group className="mb-0 text-center text-lg-right" controlId={pageSelectorId.current}>
+            <Form.Label sm={2} className="d-inline mb-0">Go to page</Form.Label>
+            <FormControl
+              defaultValue={page.toString()}
+              onKeyDown={({ key, currentTarget }: React.KeyboardEvent<HTMLInputElement>) => {
+                if (key === "Enter") {
+                  const requestedPage = Math.max(1, Math.min(parseInt(currentTarget.value), numPages)) || page;
+                  setPage(requestedPage);
+                  setDisplayPageSelector(false);
+                } else if (key === "Escape") {
+                  setDisplayPageSelector(false);
+                }
+              }}
+              onBlur={() => { setDisplayPageSelector(false) }}
```

Comment:
This might be just me, but I expected blurring the selector to also set the page.

Reply:
Took longer than I expected, but I figured out how to have an ideal submit-on-blur: 30eb79f

Reply:
Hate to nitpick some more on this, but the behaviour of the input box replacing your input with valid input as you type is kinda weird, IMO. It should only do that on submit.

Reply:
7e8a601